Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3339,12 +3339,15 @@ extension BrowserViewController: TopTabsDelegate {
         openBlankNewTabAndFocus(isPrivate: isPrivate)
     }
     
-    func topTabsDidPressPrivateTab() {
+    func topTabsDidPressPrivateTab(tab: Tab?) {
         guard let selectedTab = tabManager.selectedTab else {
             return
         }
         urlBar.leaveOverlayMode()
-        if selectedTab.isPrivate {
+        if let tab = tab {
+            tabManager.selectTab(tab)
+        }
+        else if selectedTab.isPrivate {
```

Comment:
I think "tab" was a poor choice for a parameter name, I think "cachedTab" would make more sense?  So I check if there's a cached tab passed is the function, if not then I select the most recent normal/private tab to select.